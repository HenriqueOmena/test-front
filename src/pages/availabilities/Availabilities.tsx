import React from "react";
import { useRequest } from "hooks/useRequest";
import { DoctorCard } from "components/DoctorCard";
import { IDoctorCard } from "components/DoctorCard/DoctorCard.interface";
import AvailabilitiesStyle from "./Availabilities.style";

const Availabilities: React.FC = () => {
  const { cardContainer } = AvailabilitiesStyle();
  const { apiData, isLoading } = useRequest("GET", "doctors");

  return (
    <div>
      <header className="header">
        <img src="/joovence-logo.png" className="logo" alt="logo" />
        <h1>Bookings</h1>
      </header>
      {isLoading && <span>Loading.....</span>}

      <div className={cardContainer}>
        {apiData?.map((doctor: IDoctorCard) => (
          <DoctorCard key={doctor?.id} {...doctor} />
        ))}
      </div>
    </div>
  );
};

export default Availabilities;
