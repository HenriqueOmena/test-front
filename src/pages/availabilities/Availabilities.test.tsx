import { render } from "@testing-library/react";
import React from "react";
import Availabilities from "./Availabilities";

describe("Component - Availabilities", () => {
  let component = <Availabilities />;

  it("Should Match SnapShot", () => {
    expect(component).toMatchSnapshot();
  });

  it("Should render with prop name", async () => {
    const { getByText } = render(component);
    const nameText = getByText("Bookings");

    expect(nameText).toBeVisible();
  });
});
