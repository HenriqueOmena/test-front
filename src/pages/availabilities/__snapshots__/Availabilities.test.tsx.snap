// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Availabilities Should Match SnapShot 1`] = `<Availabilities />`;
